src/commands/part.cpp

```diff
@@ -16,6 +16,8 @@ int command_part(Message &message, Dependencies &deps)
 
 	while (channel_it != channels.end())
 	{
+		if (!_base_channel::isValidName(*channel_it))
+			return 1;
```

Comment:
Should send an `403 ERR_NOSUCHCHANNEL` and skip this iteration (continue won't work as the incrementation happens at the end).